### ISSUE TITLE
Fenix potentially leaking device language?

### ISSUE THREAD

**Issue by Jee-Hex** [arkenfox/user.js#1095]
Apparently this has been going on for quite a while now but Fenix is actually overriding `intl.accept_languages` and `intl.locale.requested` on every startup to whatever option is selected under _Settings > Language_. By default this is set to _Follow device language_ so unless a user manually changes these prefs every time, they may be leaking their device locale. And even when a user sets the locale to _en-US_, `int.accept_languages` is overridden as `en-US` instead of `en-US, en`. I could manually reset `intl.accept_languages` (to `chrome://global/locale/intl.properties`) and the _Accept-Language_ header would return _en-US,en;q=0.5_ until I close fenix, but I am not sure if this would help in terms of blending in. Since Tor browser seemed to have opted for their own patch for this, what would the recommendation for RFP users on fenix beta/nightly be in the meantime?

**Comment by Thorin-Oakenpants** [comment]
Here goes
- First of all: Fenix is wildly different under the hood: there is no guarantee that prefs work the same as desktop releases
- Second: unless you are on Dev/Beta or Nightly Fenix, then about:config is not available
   - which means this is all moot given you can't flip RFP or FPI (but dFPI is not far off by default), or disable WebRTC, webGL
- Third, this repo is about desktop: as far as we're concerned: android was, and is, a privacy nightmare: not that we can't do something about it
- Fourth: it's going to get much worse re your OS language in FF85+ (I assume this applies to GeckoView)
   - [Bugzilla 1635561](https://bugzilla.mozilla.org/show_bug.cgi?id=1635561) now applies `RegionalPrefLocales` rather than `AppLocale`
   - if you're not sure what that means, it means that you **_also_** need to set the hidden pref `javascript.use_us_english_locale` as `true` (otherwise all `Intl.` uses your OS locale, formatting, resolved options etc) - and this hidden pref **_only_** fixes English (US) - so imagine trying to set it up to be French on a German OS: i.e you can't

On your phone, in Firefox you can set your language - but GeckoView only sets a single language (I think) - i.e no secondary ones (I think, or maybe it auto adds the OS language if you choose something different). It will be `en-US` (not `en-US,en` - which is what TB patched). My phone happens to have en-US as it's OS, so it's quite hard for me to test anything

I know about the TB difference, but it's not polished. And there are other issues: e.g. en-GB (great britain), en-CA (canadian) fall through the cracks. I'm biding my time before I try and re-bring-it-up with sysrqb (I did over a year ago) and upstream at Moz

tl;dr: Fenix is a bit of a mess and lacks RFP parity (not just in languages). Also since Fenix lacks about:config, nothing will be done upstream until they add a UI interface with RFP etc - and that's a long way away, since Fenix has bigger parity issues to worry about. And pestering TB devs is not a good way to wins friends and influence people

**Comment by Jee-Hex** [comment]
Thanks for the detailed explanation!

**Comment by atomGit** [comment]
check out [Privacy Browser](https://f-droid.org/en/packages/com.stoutner.privacybrowser.standard/) as a possible alternative to Fenix

**Comment by Thorin-Oakenpants** [comment]
^^ why?

Just use Fenix Nightly
